### ISSUE TITLE
Install on Ubuntu stuck at "Waiting for service pods to become ready"

### ISSUE THREAD

**Issue by saurav2608** [kubeflow/kubeflow#5181]
/kind bug

**What steps did you take and what happened:**
I am doing a fresh install on an multipass Ubuntu 18.04-LTS (on my Mac). On running `KUBEFLOW_DEBUG=true microk8s.enable kubeflow`. 

The setup is stuck at the highlighted point for 30+ mins

Enabling dns...
Enabling storage...
Enabling dashboard...
Enabling ingress...
Enabling metallb:10.64.140.43-10.64.140.49...
Waiting for DNS and storage plugins to finish setting up
Deploying Kubeflow...
Kubeflow deployed.
Waiting for operator pods to become ready.
Waited 0s for operator pods to come up, 34 remaining.
...
Waited 405s for operator pods to come up, 1 remaining.
Operator pods ready.
**_Waiting for service pods to become ready._**

**What did you expect to happen:**
Kubefow to start

**Anything else you would like to add:**
I have replicated this issue on my a cloud VM running Ubuntu 18.04 LTS. I did not use multipass but installed microk8s. 

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard):
- kfctl version: (use `kfctl version`):
- Kubernetes platform: (e.g. `minikube`)
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by cf-chiang** [comment]
I'm facing the same issue.

**What steps did you take and what happened:**
After launching a virtual machine (Ubuntu 20.04.1 LTS) on my MacBook Pro (MacOS Mojave 10.14.6) and enabling dns, dashboard and storage, I attempted to enable kubeflow with the following command: 
`KUBEFLOW_DEBUG=true microk8s.enable kubeflow`

And got the following output:

Waited 1320s for operator pods to come up, 1 remaining.
Operator pods ready.
Waiting for service pods to become ready.

It was stuck at this point for more than 2 days, and I had to interrupt it afterwards.

Output of `microk8s.inspect`
Inspecting Certificates
Inspecting services
  Service snap.microk8s.daemon-cluster-agent is running
  Service snap.microk8s.daemon-containerd is running
  Service snap.microk8s.daemon-apiserver is running
  Service snap.microk8s.daemon-apiserver-kicker is running
  Service snap.microk8s.daemon-proxy is running
  Service snap.microk8s.daemon-kubelet is running
  Service snap.microk8s.daemon-scheduler is running
  Service snap.microk8s.daemon-controller-manager is running
  Copy service arguments to the final report tarball
Inspecting AppArmor configuration
Gathering system information
  Copy processes list to the final report tarball
  Copy snap list to the final report tarball
  Copy VM name (or none) to the final report tarball
  Copy disk usage information to the final report tarball
  Copy memory usage information to the final report tarball
  Copy server uptime to the final report tarball
  Copy current linux distribution to the final report tarball
  Copy openSSL information to the final report tarball
  Copy network configuration to the final report tarball
Inspecting kubernetes cluster
  Inspect kubernetes cluster

Building the report tarball
